source/_docs/installation/synology.markdown

```diff
@@ -2,78 +2,196 @@
+bcrypt==3.1.4
+cryptography==2.3.1
 
-Use PIP to install Homeassistant package 0.64.3
+## Cross compilation requirement for installing "Warrent" module (Needed by "Cloud" Component)
+pycryptodome==3.7.2
 
-```bash
-# ./python3 -m pip install homeassistant==0.64.3
+## Cross compilation requirement for installing "HAP_python" module (Needed by "Homekit" Component)
+curve25519-donna==1.3
+ed25519==1.4
 ```
 
-Create homeassistant config directory & switch to it
-
+Add these edits to fix OpenSSL errors when using the "[Xiaomi_Aqara](/components/Xiaomi_Aqara/)" component.  
+Edit "*~/spksrc/spk/python3/Makefile*" and add the following text above the text "**include ../../mk/spksrc.spk.mk**":
+```makefile
+## Makefile variable needed to fix "_openssl.so: undefined symbol: pthread_atfork" error when using "xiaomi_aqara" component.
+export CFLAGS=-pthread
+```
+#### {% linkable_title Compiling the Python 3 package %}
+Make the Python 3 package for your Synology model, please replace "arch-**XXXX**" by the apporiate architecture of your Synology. For a list of architectures, look at this [list of architectures](https://github.com/SynoCommunity/spksrc/wiki/Architecture-per-Synology-model) accepted by spksrc. Depending on your computer, compilation may take a hour or more (Significantly less if you have a SSD and a moderately good CPU).
 ```bash
-# mkdir /volume1/homeassistant
-# chown homeassistant /volume1/homeassistant 
-# chmod 755 /volume1/homeassistant
-# cd /volume1/homeassistant
+# sudo docker run -it -v ~/spksrc:/spksrc synocommunity/spksrc /bin/bash
+$ make setup
+$ cd spk/python3
+$ make arch-XXXX
+$ exit
 ```
-Hint: alternatively you can also create a "Shared Folder" via Synology WebUI (e.g., via "File Station") - this has the advantage that the folder is visible via "File Station".
-
-Create hass-daemon file using the following code (edit the variables in uppercase if necessary)
+After the compilation is done, you can find the Python 3 package at "*~/spksrc/packages/python3_XXXX.spk*".
+It should be named something like "python3_armada370-6.1_3.5.5-7.spk", of course with a possibly different architecture and version.
 
+#### {% linkable_title Extracting cross compiled packages %}
+Now you need to extract the Python module packages which you added earlier to "*requirements.txt*".
+Run these commands to extract the `.whl` files to a directory the "**~/Module-Packages**", please modify `pyspk` command so the "**XXXX**" in `python3_XXXX.spk` points to the package file you made earlier.
+```bash
+$ mkdir ~/Module-Packages
+$ cd ~/Module-Packages
+$ pyspk="$HOME/spksrc/packages/python3_XXXX.spk"
+$ tar -x -f $pyspk -C /tmp package.tgz; gzip -df /tmp/package.tgz
+$ tar -x -f /tmp/package.tar --wildcards share/wheelhouse/$file --strip=2
+```
+Inside some of the `.whl` files (These are zip archives) you need to rename all files containing the text "**x86_64-linux-gnu**" to "**arm-linux-gnueabihf**", this is only required for ARM based Synology's.  
```

Comment:
Synology DSMs would refer to the operating system, which in this case seems to be the most fitting.
English is not my native language, but looking at some noun rules, should "DSMs" not be "DSMs'"?

Reply:
You're correct, it would be "Synology DiskStations" or "Synology Products".  This is like the "Lego-vs-Legos" argument, where "Legos" is not actually plural of "Lego": it would be "Lego Bricks", "Lego toys", or just "Lego" (pluralizing like "sheep", "moose", "fish")

Synology public relations refers to their products as "Synology Products" or "Synology NAS" (I guess using "NAS" as plural for "NAS" :) )

Reply:
NAS is more user friendly, DSMs' sounds weird, I will be taking NAS.

Reply:
I decided to remove the "#Extracting cross compiled packages" part since after a little testing I saw that installing the `.whl` is not needed, which means the edits discussed in this one are invalidated.
Must have left it around because in the beginning I installed Home Assistant in virtual python environment.

Oh well...